code/modules/research/xenobiology/vatgrowing/vatgrower.dm

```diff
@@ -170,6 +174,39 @@
 		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), get_turf(src), 'sound/effects/servostep.ogg', 100, 1), 1.5 SECONDS)
 		biological_sample.reset_sample()
 	else
-		UnregisterSignal(biological_sample, COMSIG_SAMPLE_GROWTH_COMPLETED)
+		UnregisterSignal(biological_sample, list(COMSIG_SAMPLE_GROWTH_COMPLETED, COMSIG_SAMPLE_DEPOSITED))
 		QDEL_NULL(biological_sample)
 	update_appearance()
+
+/obj/machinery/vatgrower/proc/on_sample_deposited(datum/biological_sample/sample, atom/thing)
+	SIGNAL_HANDLER
```

Comment:
Why is this registered on the base instead of the subtype if only the subtype implements it anyways